src/VisualStudio/IntegrationTest/New.IntegrationTests/VisualBasic/BasicRename.cs

```diff
@@ -0,0 +1,401 @@
+            AssertEx.SetEqual(renameSpans, tagSpans);
+
+            await TestServices.Input.SendWithoutActivateAsync("Custom", HangMitigatingCancellationToken);
+            await TestServices.Workspace.WaitForRenameAsync(HangMitigatingCancellationToken);
+            await TestServices.EditorVerifier.TextEqualsAsync(@"
+Imports System
+
+<Custom>
+Class Bar
+End Class
+
+Public Class Custom$$Attribute 
+        Inherits Attribute
+End Class", HangMitigatingCancellationToken);
+        }
+
+        [IdeFact, WorkItem("https://github.com/dotnet/roslyn/issues/21657")]
+        public async Task VerifyAttributeCapitalizedRename()
+        {
+            var markup = @"
+Imports System
+
+Public Class [|$$ustom|]ATTRIBUTE
+        Inherits Attribute
+End Class";
+            await SetUpEditorAsync(markup, HangMitigatingCancellationToken);
+            await TestServices.InlineRename.InvokeAsync(HangMitigatingCancellationToken);
+
+            MarkupTestFile.GetSpans(markup, out var _, out ImmutableArray<TextSpan> renameSpans);
+            var tags = await TestServices.Editor.GetRenameTagsAsync(HangMitigatingCancellationToken);
+            var tagSpans = tags.SelectAsArray(tag => new TextSpan(tag.Span.Start, tag.Span.Length));
+            AssertEx.SetEqual(renameSpans, tagSpans);
+
+            await TestServices.Input.SendWithoutActivateAsync(new InputKey[] { "Custom", VirtualKeyCode.RETURN }, HangMitigatingCancellationToken);
+            await TestServices.Workspace.WaitForRenameAsync(HangMitigatingCancellationToken);
+            await TestServices.EditorVerifier.TextEqualsAsync(@"
+Imports System
+
+Public Class CustomAttribute$$
+    Inherits Attribute
+End Class", HangMitigatingCancellationToken);
+        }
+
+        [IdeFact, WorkItem("https://github.com/dotnet/roslyn/issues/21657")]
+        public async Task VerifyAttributeNotCapitalizedRename()
+        {
+            var markup = @"
+Imports System
+
+Public Class [|$$ustom|]attribute
+        Inherits Attribute
+End Class";
+            await SetUpEditorAsync(markup, HangMitigatingCancellationToken);
+            await TestServices.InlineRename.InvokeAsync(HangMitigatingCancellationToken);
+
+            MarkupTestFile.GetSpans(markup, out var _, out ImmutableArray<TextSpan> renameSpans);
+            var tags = await TestServices.Editor.GetRenameTagsAsync(HangMitigatingCancellationToken);
+            var tagSpans = tags.SelectAsArray(tag => new TextSpan(tag.Span.Start, tag.Span.Length));
+            AssertEx.SetEqual(renameSpans, tagSpans);
+
+            await TestServices.Input.SendWithoutActivateAsync(new InputKey[] { "Custom", VirtualKeyCode.RETURN }, HangMitigatingCancellationToken);
+            await TestServices.Workspace.WaitForRenameAsync(HangMitigatingCancellationToken);
+            await TestServices.EditorVerifier.TextEqualsAsync(@"
+Imports System
+
+Public Class CustomAttribute$$
```

Comment:
```suggestion
Public Class CustomA$$ttribute
```

This is what the actual result looks like in the failure

Reply:
I tested Version 17.6.0 Preview 3.0 [33505.110.main] both with and without this change, and so far I have not been able to reproduce a caret position anywhere except at the end of the line.

Reply:
This is what I got in 17.6p3
![rename](https://user-images.githubusercontent.com/788783/225178900-f0a6e7aa-3299-4517-b08d-6d11192c4ba0.gif)

Reply:
That's a different test. The one here is only the behavior for attributes.

Reply:
I modified the test to support both caret placement outcomes. It's not clear how/why the caret gets placed at the end of a rename operation so I wasn't sure how to start debugging this.